arduino-ide-extension/src/browser/contributions/board-selection.ts

```diff
@@ -79,22 +80,83 @@ export class BoardSelection extends SketchContribution {
-          const detail = `BN: ${selectedBoard.name}
-VID: ${VID}
-PID: ${PID}`;
-          await remote.dialog.showMessageBox(remote.getCurrentWindow(), {
-            message: nls.localize('arduino/board/boardInfo', 'Board Info'),
-            title: nls.localize('arduino/board/boardInfo', 'Board Info'),
-            type: 'info',
-            detail,
-            buttons: [nls.localize('vscode/issueMainService/ok', 'OK')],
-          });
+        // IDE2 must show the board info based on the selected port.
+        // https://github.com/arduino/arduino-ide/issues/1489
+        // IDE 1.x does not support network ports
+        if (selectedPort.protocol === 'network') {
+          this.messageService.info(
+            nls.localize(
+              'arduino/board/noNetworkPort',
+              "Network port, can't obtain info."
+            )
+          );
+          return;
+        }
+        // serial protocol with one or many detected boards or available VID+PID properties from the port
+        const isNonNativeSerial = (port: Port, boards: Board[]) => {
+          return (
+            port.protocol === 'serial' &&
+            (boards.length ||
+              (port.properties?.['vid'] && port.properties?.['pid']))
+          );
+        };
+        const selectedPortKey = Port.keyOf(selectedPort);
+        const state = await this.boardsService.getState();
+        const boardListOnSelectedPort = Object.entries(state).filter(
+          ([portKey, [port, boards]]) =>
+            portKey === selectedPortKey && isNonNativeSerial(port, boards)
+        );
+        // IDE 1.x show this when cannot identify for example a ESP8266MOD although compile and upload works
+        if (!boardListOnSelectedPort.length) {
+          this.messageService.info(
+            nls.localize(
+              'arduino/board/noNativeSerialPort',
+              "Native serial port, can't obtain info."
+            )
+          );
+          return;
         }
+
+        const [, [port, boards]] = boardListOnSelectedPort[0];
+        if (boardListOnSelectedPort.length > 1 || boards.length > 1) {
+          console.warn(
+            `Detected more than one available boards on the selected port : ${JSON.stringify(
+              selectedPort
+            )}. Detected boards were: ${JSON.stringify(
+              boardListOnSelectedPort
+            )}. Using the first one: ${JSON.stringify([port, boards])}`
+          );
+        }
+
+        const board = boards[0];
+        const falsyToNullString = (s: string | undefined) =>
+          !!s ? s : '(null)';
+        const readProperty = (property: string, port: Port) =>
+          falsyToNullString(port.properties?.[property]);
+        const BN =
+          board?.name ??
+          nls.localize('arduino/board/unknownBoard', 'Unknown board');
+        const VID = readProperty('vid', port);
+        const PID = readProperty('pid', port);
+        const SN = board
+          ? readProperty('serialNumber', port)
```

Comment:
>For the unidentified board, the SN is available: AB0P8H9K. Is this correct to show: "Upload any sketch to obtain it."?

No. If the port has a `serialNumber` property, that should always be shown in the"**SN**" field of the "**Board Info**" dialog.

The board might only be unidentified because the platform is not installed (in combination with no "builder API" access in the case of official boards, as would be the case if the computer isn't connected to the Internet).